src/events/SDL_events.c

```diff
@@ -973,7 +973,10 @@ SDL_WaitEventTimeout_Device(_THIS, SDL_Window *wakeup_window, SDL_Event * event,
             status = _this->WaitEventTimeout(_this, loop_timeout);
             /* Set wakeup_window to NULL without holding the lock. */
             _this->wakeup_window = NULL;
-            if (status <= 0) {
+            if (status == 0 && loop_timeout == PERIODIC_POLL_INTERVAL_MS) {
```

Comment:
Shouldn't this be `if (need_periodic_poll)`? The timeout could happen to be `PERIODIC_POLL_INTERVAL_MS` if it lines up just right.

Reply:
Yeah, it doesn't hurt if we loop again if the callers passes `PERIODIC_POLL_INTERVAL_MS` since we'll exit in the `elapsed >= (Uint32)timeout` check above, but no sense in doing that if we're not using the periodic poll anyway.